### PR TITLE
Op refine for making build system more automatic

### LINKED ISSUE #3872: Cmake warning on develop branch
On current develop branch, when run `cmake` it prints the following warning message:

```
CMake Warning at paddle/operators/CMakeLists.txt:35 (message):
  The op library net_op not support GPU!
Call Stack (most recent call first):
  paddle/operators/CMakeLists.txt:56 (op_library)
```

~~Maybe related with: https://github.com/PaddlePaddle/Paddle/pull/3801~~

### PR DESCRIPTION
**Pull request by luotao1**
fix https://github.com/PaddlePaddle/Paddle/issues/3807#issuecomment-326549728
fix #3872 : As there will be more only-CPU ops, it is not necessary to message these lines.
related #3852 .

### CODE REVIEW

**Comment by luotao1** [comment]
@typhoonzero As the commit info of #3852 is mass, I create a new PR, and all your commens are Done.

**Comment by typhoonzero** [comment]
Seem ci failed at: 

```
CMakeFiles/grad_op_builder_test.dir/grad_op_builder_test.cc.o: In function `__static_initialization_and_destruction_0(int, int) [clone .constprop.318]':
[20:05:40]	grad_op_builder_test.cc:(.text.startup+0x86): undefined reference to `TouchOpKernelRegistrar_add_GPU()'
[20:05:40]	collect2: error: ld returned 1 exit status
```

**Comment by luotao1** [comment]
I see it, I am trying to fix it.

**Comment by luotao1** [comment]
fix the ci error, and remove unnecessary `.h` for `add_op.cu`

File: paddle/operators/scale_op.cc [hunk 1]
```diff
@@ -65,33 +65,6 @@ class ScaleGradOp : public NetOp {
   }
 };
 
-// identity is a alias of scale op. This is also a example for creating a alias
-// operator.
-template <typename AttrType>
-class IdentityOpMaker : public framework::OpProtoAndCheckerMaker {
```
Comment (reyoung) [inline paddle/operators/scale_op.cc]: Why remove the `identity_op`?

Comment (luotao1) [reply paddle/operators/scale_op.cc]: @reyoung 只是把 `identity_op`弄成了一个单独的文件，没有放在`scale_op`里面，并没有删掉它。这样CMakeLists.txt能自动根据`operators/`目录下的文件名，来获得有哪些op需要编译，之后也方便自动生成`pybind.cc`中的`USE_OP(XXX)`，而不用一个个手写添加。

File: paddle/operators/CMakeLists.txt [hunk 2]
```diff
@@ -14,27 +14,31 @@ function(op_library TARGET)
...
 
     list(LENGTH cc_srcs cc_srcs_len)
     if (${cc_srcs_len} EQUAL 0)
         message(FATAL_ERROR "The op library ${TARGET} should contains at least one .cc file")
     endif()
 
-    list(LENGTH cu_srcs cu_srcs_len)
-    list(LENGTH op_library_DEPS dep_len)
-    if (${cu_srcs_len} EQUAL 0 AND ${dep_len} EQUAL 0)
-        message(WARNING "The op library ${TARGET} not support GPU!")
```
Comment (typhoonzero) [inline paddle/operators/CMakeLists.txt]: Remove message is fine, because sometimes we can implement the CPU version only and then implement GPU version.

**Pull request merged**